ui/index.js

```diff
@@ -129,10 +131,40 @@ export async function setupInitialStore(
 
   if (getEnvironmentType() === ENVIRONMENT_TYPE_POPUP) {
     const { origin } = draftInitialState.activeTab;
+
+    // Not entirely sure if these changes are correct. The original logic kind of confuses me.
     const permittedAccountsForCurrentTab =
-      getPermittedAccountsForCurrentTab(draftInitialState);
-    const selectedAddress =
-      getSelectedInternalAccount(draftInitialState)?.address ?? '';
+      getAllPermittedAccountsForCurrentTab(draftInitialState);
+
+    const selectedAccount = getSelectedInternalAccount(draftInitialState);
```

Comment:
i guess i'm mainly not understanding why the nullish coasce is needed above

Reply:
You mean why the nullish coasce **was** needed above before your changes?